pkg/operator/operator.go

```diff
@@ -405,6 +405,37 @@ func New(
 	}
 	o.controllersToRunFunc = append(o.controllersToRunFunc, csrController.Run, o.ruleController.Run, o.relabelController.Run)
 
+	// csrFederateController runs a controller that requests a client TLS certificate for the telemeter client. This certificate is used to authenticate against the Prometheus /federate API endpoint.
+	csrFederateController, err := csr.NewClientCertificateController(
+		csr.ClientCertOption{
+			SecretNamespace: "openshift-monitoring",
+			SecretName:      "federate-client-certs",
+		},
+		csr.CSROption{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "system:openshift:openshift-monitoring-",
+				Labels: map[string]string{
+					"metrics.openshift.io/csr.subject": "prometheus",
+				},
+			},
```

Comment:
I'd try with an empty CSROption, I don't think that it is used anywhere by the controller.

```suggestion
		csr.CSROption{},
```

Reply:
It returns an error.:
```
E0602 11:43:35.715902  131677 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory addr
ess or nil pointer dereference)            
```

Reply:
hmm do you have the full stack trace?